test_fms/topography/test_topography.F90

```diff
@@ -0,0 +1,310 @@
+
+    !---------------------------------------- test get_ocean_mask 1d ---------------------------------------------!
+
+    get_ocean_mask_answer = get_ocean_mask(lon1d, lat1d, ocean_mask1d)
+
+    if (get_ocean_mask_answer .neqv. .true.) call mpp_error(FATAL, "ocean field not read correctly")
+    if (ocean_mask1d(1,1) .neqv. .false.) call mpp_error(FATAL, "test_get_ocean_mask 1d: ocean mask should be false")
+
+  end subroutine test_get_ocean_mask
+
+  subroutine test_get_water_frac
+
+    implicit none
+    real(kind=TEST_TOP_KIND_), dimension(size(lon2d,1)-1,size(lat2d,2)-1) :: water_frac2d
+    real(kind=TEST_TOP_KIND_), dimension(size(lon1d)-1,size(lat1d)-1)     :: water_frac1d
+    logical                                                               :: get_water_frac_answer
+
+    !---------------------------------------- test get_water_frac 2d ---------------------------------------------!
+
+    get_water_frac_answer = get_water_frac(lon2d, lat2d, water_frac2d)
+
+    if (get_water_frac_answer .neqv. .true.) call mpp_error(FATAL, "ocean field not read correctly")
+    call check_answers(water_frac2d(1,1), 0.5_lkind, "Error in test_get_water_frac 2d")
+
+    !---------------------------------------- test get_water_frac 1d ---------------------------------------------!
+
+    get_water_frac_answer = get_water_frac(lon1d, lat1d, water_frac1d)
+
+    if (get_water_frac_answer .neqv. .true.) call mpp_error(FATAL, "ocean field not read correctly")
+    call check_answers(water_frac1d(1,1), 0.5_lkind, "Error in test_get_ocean_frac 1d")
+
+  end subroutine test_get_water_frac
+
+  subroutine test_get_water_mask
+
+    implicit none
+    logical, dimension(size(lon2d,1)-1,size(lat2d,2)-1) :: water_mask2d
+    logical, dimension(size(lon1d)-1,size(lat1d)-1)     :: water_mask1d
+    logical                                             :: get_water_mask_answer
+
+    !---------------------------------------- test get_water_mask 2d ---------------------------------------------!
+
+    get_water_mask_answer = get_water_mask(lon2d, lat2d, water_mask2d)
+
+    if (get_water_mask_answer .neqv. .true.) call mpp_error(FATAL, "ocean field not read correctly")
+    if (water_mask2d(1,1) .neqv. .false.) call mpp_error(FATAL, "test_get_water_mask 2d: ocean mask should be false")
+
+    !---------------------------------------- test get_water_mask 1d ---------------------------------------------!
+
+    get_water_mask_answer = get_ocean_mask(lon1d, lat1d, water_mask1d)
+
+    if (get_water_mask_answer .neqv. .true.) call mpp_error(FATAL, "ocean field not read correctly")
+    if (water_mask1d(1,1) .neqv. .false.) call mpp_error(FATAL, "test_get_ocean_mask 1d: ocean mask should be false")
+
+  end subroutine test_get_water_mask
+
+  subroutine check_answers(result, answer, what_error)
+
+    implicit none
+    real(kind=TEST_TOP_KIND_) :: result       ! value calculated from script
```

Comment:
result is a Fortran keyword.  Could another variable name be used here?

Reply:
i should totally know that huh? lol